stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/DtoMapper.kt

```diff
@@ -0,0 +1,59 @@
+package io.getstream.chat.android.client.api2
+
+import io.getstream.chat.android.client.api2.model.dto.DownstreamMessageDto
+import io.getstream.chat.android.client.api2.model.dto.UpstreamMessageDto
+import io.getstream.chat.android.client.api2.model.dto.UserDto
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+
+internal class DtoMapper {
+
+    fun toDto(message: Message): UpstreamMessageDto = message.run {
+        UpstreamMessageDto(
+            id = id,
+            cid = cid,
+            text = text,
+            html = html,
+            parent_id = parentId,
+            command = command,
+            user = toDto(user),
+            silent = silent,
+            shadowed = shadowed,
+        )
+    }
+
+    fun toDomain(message: DownstreamMessageDto): Message = message.run {
+        return Message(
+            id = id,
+            cid = cid,
+            text = text,
+            html = html,
+            parentId = parent_id,
+            command = command,
+            user = toDomain(user),
+            silent = silent,
+            shadowed = shadowed,
+            createdAt = created_at,
+            updatedAt = updated_at,
+            deletedAt = deleted_at,
+        )
+    }
+
+    fun toDto(user: User): UserDto = user.run {
```

Comment:
What about fun extensions?? :thinking: 
```
internal fun User.toDto(): UserDto
```

Reply:
I think I'll go with this approach, but I'll do it in one of the next PRs.